cranelift-codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -212,12 +246,19 @@ impl<'builder> Template<'builder> {
         copy
     }
     pub fn nonrex(&self) -> Self {
-        assert!(!self.requires_prefix, "Tail recipe requires REX prefix.");
+        assert!(
+            self.rex_kind != RexRecipeKind::AlwaysEmitRex,
+            "Tail recipe requires REX prefix."
```

Comment:
oops, preexisting: these are called Templates since the migration to Rust, not TailRecipes anymore. (a few times below too)